cal-itp/benefits
# Refactor: move EligibilityType fields to EnrollmentFlow

Closes #2283

- [x] Fields from `EligibilityType` are moved to `EnrollmentFlow`
- [x] Migrate the data from `EligibilityType.name` to `EnrollmentFlow.name` in the migration file
- [x] The `label` field is used as the Admin display field for `EnrollmentFlow` instances
- [x] `TransitAgency.eligibility_types` field (and all related calculated fields) are removed
- [x] `EligibilityType` model is removed
- [x] `session.eligibility_types` is removed
- [x] `session.eligible` (get `bool`) used to indicate if the user was found eligible for their selected `flow`, updated through `session.update(eligible=bool)` like other session data
- [x] [`EligibilityType` read-only fields](https://github.com/cal-itp/benefits/blob/main/benefits/core/admin.py#L55) are moved to the new model
- [x] Analytics events that currently include `eligibility_types` properties (event/user) now use `enrollment_flows.eligibility_name` for the `eligibility_types` property values

## Testing locally

1. Checkout the `main` branch
1. Make sure that your `local_fixtures.json` (or whatever fixture you are using) has an `EnrollmentFlow` that has a foreign key reference to an `EligibilityType`.
1. Run `bin/reset_db.sh` (now your local DB mirrors the structure of the `dev` environment)
1. Checkout this PR's branch
1. Run `bin/init.sh` (running this PR's migration on top of existing DB structure)
